src/zenml/pipelines/pipeline_decorator.py

```diff
@@ -80,6 +87,15 @@ def pipeline(
         enable_artifact_metadata: Whether to enable artifact metadata or not.
         settings: Settings for this pipeline.
         extra: Extra configurations for this pipeline.
+        on_failure: Callback function in event of failure of the step. Can be
+            a function with three possible parameters,
+            `StepContext`, `BaseParameters`, and `Exception`,
+            or a UDF path to a function of the same specifications
```

Comment:
What is a UDF path?

Reply:
user-defined function: https://en.wikipedia.org/wiki/User-defined_function

Reply:
Ah I see, yes that needs more explanation. To be consistent with other places, we might also just want to call it `source path` or `source`